subsys/net_core_monitor/app_core.c

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include <init.h>
+#include <errno.h>
+#include <kernel.h>
+#include <sys/printk.h>
+
+#include <logging/log.h>
+#include <nrfx_ipc.h>
+
+#define IPC_MEM_IDX 0
+
+LOG_MODULE_REGISTER(net_core_monitor, CONFIG_NET_CORE_MONITOR_LOG_LEVEL);
+
+static uint32_t prev_reset_cnt;
+
+static void net_reset(void)
+{
+	LOG_WRN("Resetting net");
+	NRF_RESET->NETWORK.FORCEOFF = RESET_NETWORK_FORCEOFF_FORCEOFF_Hold;
+	k_busy_wait(10);
+	NRF_RESET->NETWORK.FORCEOFF = RESET_NETWORK_FORCEOFF_FORCEOFF_Release;
+}
+
+int ncm_net_status_check(bool reset_on_failure)
+{
+	int ret;
+	uint32_t mem;
+	uint32_t diff;
+	static uint32_t prev_cnt;
+
+	mem = nrf_ipc_gpmem_get(NRF_IPC, IPC_MEM_IDX);
+
+	uint32_t cnt = mem & 0xFFFF;
+	uint32_t reset_cnt = mem >> 16;
+
+	LOG_DBG("mem: 0x%08X", mem);
+
+	if (reset_cnt != prev_reset_cnt) {
+		prev_reset_cnt = reset_cnt;
+		ret = -EFAULT;
+	} else {
+		diff = (cnt - prev_cnt) & BIT_MASK(16);
+		if (diff == 0 || diff > 3) {
```

Comment:
Why 3? If this function was called from a thread with low priority Net Core could increment its counter much more than 3 between two invocations of this function.

Reply:
The feeding interval is long. If app cannot schedule threat one can increase the interval.

Still you are right as I don't see any reason we have `3` as a limit. I think this was added to detect the overflow but it should be done neater. Maybe `> UINT16_MAX` or maybe simply use `uint16_t` for cnt and prev_cnt and let overflow be handled within type.
FYI  @maje-emb .